freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/StackService.java

```diff
@@ -86,14 +86,25 @@ public Stack getByEnvironmentCrnAndAccountId(String environmentCrn, String accou
                 .or(() -> childEnvironmentService.findParentByEnvironmentCrnAndAccountId(environmentCrn, accountId));
     }
 
-    public Stack getByEnvironmentCrnAndAccountIdEvenIfTerminated(String environmentCrn, String accountId) {
-        return findByEnvironmentCrnAndAccountIdEvenIfTerminated(environmentCrn, accountId)
-                .orElseThrow(() -> new NotFoundException(String.format("FreeIPA stack by environment [%s] has never existed", environmentCrn)));
+    public Optional<Stack> findByCrnAndAccountIdWithListsEvenIfTerminated(String environmentCrn, String accountId, String crn) {
+        return stackRepository.findByAccountIdEnvironmentCrnAndCrnWithListsEvenIfTerminated(environmentCrn, accountId, crn)
+                .or(() -> childEnvironmentService.findParentStackByChildEnvironmentCrnAndCrnWithListsEvenIfTerminated(environmentCrn, accountId, crn));
     }
 
-    public Optional<Stack> findByEnvironmentCrnAndAccountIdEvenIfTerminated(String environmentCrn, String accountId) {
-        return stackRepository.findByEnvironmentCrnAndAccountIdEvenIfTerminated(environmentCrn, accountId)
-                .or(() -> childEnvironmentService.findParentByEnvironmentCrnAndAccountId(environmentCrn, accountId));
+    public List<Stack> getMultipleByEnvironmentCrnAndAccountIdEvenIfTerminated(String environmentCrn, String accountId) {
```

Comment:
@doktoric This affects existing customer, return was `Optional<Stack>` and now it returns `List<Stack>`. However this should only return 0 or 1 entry as long as the customer hasn't run the rebuild logic which is only callable via `curl` right now (customer can't invoke it).